plugins/circleci/src/circleci-config.ts

```diff
@@ -1,29 +1,178 @@
-import { promises as fs } from 'fs'
-import merge from 'lodash/merge'
-import path from 'path'
-import YAML from 'yaml'
 import { ToolKitError } from '@dotcom-tool-kit/error'
 import { styles } from '@dotcom-tool-kit/logger'
 import { getOptions } from '@dotcom-tool-kit/options'
 import { Hook } from '@dotcom-tool-kit/types'
+import { automatedComment, CircleConfig, Job, JobConfig, Workflow } from '@dotcom-tool-kit/types/lib/circleci'
 import { semVerRegex } from '@dotcom-tool-kit/types/lib/npm'
-import { Workflow, JobConfig, CircleConfig, automatedComment, Job } from '@dotcom-tool-kit/types/lib/circleci'
+import type { CircleCIOptions } from '@dotcom-tool-kit/types/lib/schema/circleci'
+import { promises as fs } from 'fs'
+import isMatch from 'lodash/isMatch'
+import merge from 'lodash/merge'
+import mergeWith from 'lodash/mergeWith'
+import path from 'path'
+import type { PartialDeep } from 'type-fest'
+import YAML from 'yaml'
+
+const majorOrbVersion = '3'
 
-const majorOrbVersion = '2'
+export type CircleCIState = CircleConfig
```

Comment:
**question:** why are you creating a new type here? Why not use just `CircleConfig`? Because `CircleCIState` changes as the config functions are run?

Reply:
I've been exporting a `*State` type for [each plugin](https://github.com/Financial-Times/dotcom-tool-kit/blob/69904edc5e01f484e836289932c5378ba9ec9f36/lib/package-json-hook/src/package-json-helper.ts#L18) that uses the shared install state, in case other packages want to reference the state type. `CircleCIState` was previously a [custom interface](https://github.com/Financial-Times/dotcom-tool-kit/blob/69904edc5e01f484e836289932c5378ba9ec9f36/plugins/circleci/src/circleci-config.ts#L14), but now happens to be an alias to another type. In the future we could add additional fields to `CircleCIState` type that are not included in the `CircleConfig` type without causing a breaking change.